tests/foreman/ui/test_host.py

```diff
@@ -231,10 +231,24 @@ def test_positive_end_to_end(session, module_global_params, target_sat, host_ui_
         assert displayed_host['Installed'] == 'N/A'
         # update
         session.host.update(host_name, {'host.name': new_name})
-        assert not session.host.search(host_name)
+        assert session.host.search(host_name)[0][0] == 'No Results'
         assert session.host.search(new_host_name)[0]['Name'] == new_host_name
         # delete
-        session.host.delete(new_host_name)
+        headers = session.all_hosts.get_displayed_table_headers()
+        stripped_headers = tuple(
+            header for header in headers if header is not None and header != 'Name'
+        )
+        wait_for(lambda: session.browser.refresh(), timeout=5)
+        # Make sure there is only Name column displayed
+        session.all_hosts.manage_table_columns({header: False for header in stripped_headers})
+
+        @request.addfinalizer
+        def _finalize():
+            # Get table to original state
+            wait_for(lambda: session.browser.refresh(), timeout=5)
```

Comment:
Hmmm, it looks like that in tear down the session has no browser set. :(
```
15:15:11      >   wait_for(lambda: session.browser.refresh(), timeout=5)
15:15:11                           ^^^^^^^^^^^^^^^^^^^^^^^
15:15:11      E   AttributeError: 'NoneType' object has no attribute 'refresh'
```
So the next line will probably fail too. Is there another way (API/CLI) to set the columns back to default?

Reply:
Yeah I implemented that wrong. It should work now :D